documentation/managing_modules.md

```diff
@@ -0,0 +1,332 @@
+`bolt-project.yaml` or `Puppetfile`. To resolve the conflict and install the
+modules declared in `bolt-project.yaml`, run the following command:
+
+*\*nix shell command*
+```shell
+bolt module install --force 
+```
+
+*Powershell cmdlet*
+```shell
+Install-BoltModule -Force
+```
+
+## Update your project's modules
+
+To update the modules in your project, run:
+
+*\*nix shell command*
+```shell
+bolt module install
+```
+
+*Powershell cmdlet*
+```shell
+Install-BoltModule
+```
+
+Bolt updates your modules to the latest available versions and attempts to
+resolve all of the modules in the project configuration file to ensure that
+there are no version conflicts between modules and their dependencies.
+
+## Pin a module version
+
+If you need to pin a module in your Bolt project to a specific version, you can
+add a version requirement to the module in your `bolt-project.yaml` and run the
+`install` command with the `force` option.
+
+Follow these steps to pin a module version:
+1. Find the module under the `modules` key in your `bolt-project.yaml` file.
+2. Add a `version_requirement` key that specifies the version requirements for
+   the module. For example, the following `bolt-project.yaml` file sets the
+   version requirement for the `apache` module to greater than or equal to
+   4.0.0, but less than 6.0.0:
+   ```yaml
+   # bolt-project.yaml
+   ...
+   modules:
+   - name: puppetlabs/apache
+     version_requirement: '>= 4.0.0 < 6.0.0'
+   ```
+3. Run the following command. The `force` option is required because
+   you've made a change to your `bolt-project.yaml` file, and it no longer
+   matches the Puppetfile.
+   
+   *\*nix shell command*
+   ```shell
+   bolt module install --force 
+   ```
+
+   *Powershell cmdlet*
+   ```shell
+   Install-BoltModule -Force
+   ```
+
+When you run the install command with the `force` option, Bolt attempts to resolve all of the
+modules in the project configuration file to ensure that there are no version
+conflicts between modules and their dependencies. Bolt fails with an error
+message if it cannot resolve a dependency due to a version requirement. 
+
+> 🔩 **Tip**: For information on how to specify module versions, see the Puppet
+> documentation on [Specifying
+> versions](https://puppet.com/docs/puppet/latest/modules_metadata.html#specifying-versions).
+
+## Compatibility with Bolt versions
+
+The new module management style is incompatible with older Bolt versions, since it sets the
+moduledir in the Puppetfile to `.modules/` which is not on the modulepath in versions < 2.30.0. If
+you're using the new module management system in an environment in Puppet Enterprise, you need to
+specify `.modules` on the modulepath in your [environment.conf](https://puppet.com/docs/puppet/latest/config_file_environment.html#example), like so:
+
+```
+# /etc/puppetlabs/code/environments/test/environment.conf
+
+modulepath = .modules:site:dist:modules:$basemodulepath
```

Comment:
Won't this cause `.modules` to take precedence over every other directory?

Reply:
Woops yes.